packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -0,0 +1,895 @@
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        preventHandleOutsideImage: false,
+        spline: {
+          configuration: {
+            CARDINAL: {
+              Class: CardinalSpline,
+              scale: 0.5,
+            },
+            CATMULLROM: {
+              Class: CatmullRomSpline,
+            },
+            LINEAR: {
+              Class: LinearSpline,
+            },
+            BSPLINE: {
+              Class: BSpline,
+              addControlPointEnabled: false,
+              deleteControlPointEnabled: false,
+              showControlPointsConnectors: true,
+            },
+            DEFAULT: {
+              resolution: 20,
+              addControlPointLineDistance: 6,
+              delControlPointDistance: 6,
+              addControlPointEnabled: true,
+              deleteControlPointEnabled: true,
+              showControlPointsConnectors: false,
+            },
+          },
+          type: 'CATMULLROM',
+        },
+        actions: [
+          {
+            method: 'addControlPoint',
+            bindings: [
+              {
+                mouseButton: MouseBindings.Primary,
+                modifierKey: KeyboardBindings.Shift,
+              },
+            ],
+          },
+          {
+            method: 'deleteControlPoint',
+            bindings: [
+              {
+                mouseButton: MouseBindings.Primary,
+                modifierKey: KeyboardBindings.Ctrl,
+              },
+            ],
+          },
+        ],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+
+    this._throttledCalculateCachedStats = throttle(
+      this._calculateCachedStats,
+      100,
+      { trailing: true }
+    );
+  }
+
+  private _getSplineConfig(type: string) {
+    const { configuration: config } = this;
+    const splineConfigs = config.spline.configuration;
+    const defaultConfig = splineConfigs['DEFAULT'];
+
+    return Object.assign({ type }, defaultConfig, splineConfigs[type]);
+  }
+
+  private _getActiveSplineConfig() {
+    const { type } = this.configuration.spline;
+    return this._getSplineConfig(type);
+  }
+
+  /**
+   * Based on the current position of the mouse and the current imageId to create
+   * a CircleROI Annotation and stores it in the annotationManager
+   *
+   * @param evt -  EventTypes.NormalizedMouseEventType
+   * @returns The annotation object.
+   *
+   */
+  addNewAnnotation = (
+    evt: EventTypes.InteractionEventType
+  ): SplineROIAnnotation => {
+    const eventDetail = evt.detail;
+    const { currentPoints, element } = eventDetail;
+    const worldPos = currentPoints.world;
+
+    const enabledElement = getEnabledElement(element);
+    const { viewport, renderingEngine } = enabledElement;
+
+    this.isDrawing = true;
+
+    const camera = viewport.getCamera();
+    const { viewPlaneNormal, viewUp } = camera;
+    const splineConfig = this._getActiveSplineConfig();
```

Comment:
_getActiveSplineConfig is uses only here, how about we just put the logic here instead of refactoring the function, it is one line